cdisc_rules_engine/services/cache/cache_populator_service.py

```diff
@@ -0,0 +1,251 @@
+import asyncio
+import pickle
+from datetime import datetime, timedelta
+from functools import partial
+from typing import Iterable, List, Optional
+
+from cdisc_library_client.custom_exceptions import (
+    ResourceNotFoundException as LibraryResourceNotFoundException,
+)
+
+from cdisc_rules_engine.enums.library_endpoints import LibraryEndpoints
+from cdisc_rules_engine.services.cache.cache_service_interface import (
+    CacheServiceInterface,
+)
+from cdisc_rules_engine.services.cdisc_library_service import CDISCLibraryService
+from cdisc_rules_engine.utilities.utils import (
+    get_library_variables_metadata_cache_key,
+    get_standard_details_cache_key,
+)
+
+
+class CachePopulator:
+    def __init__(self, cache: CacheServiceInterface, api_key: str):
```

Comment:
I would suggest to make CDISCLibraryService instance an input parameter to inject this dependency.